core/templates/dev/head/domain/exploration/StateObjectFactory.js

```diff
@@ -19,36 +19,47 @@
 
 oppia.factory('StateObjectFactory', [
   'AnswerGroupObjectFactory', 'InteractionObjectFactory',
-  function(AnswerGroupObjectFactory, InteractionObjectFactory) {
-  var State = function(name, classifierModelId, contentBackendList,
-    interactionBackendDict, paramChangesBackendList) {
-    this.name = name;
-    this.classifierModelId = classifierModelId;
-    this.content = contentBackendList;
-    this.interaction = InteractionObjectFactory.create(interactionBackendDict);
-    this.paramChanges = paramChangesBackendList;
-  };
+  'ContentObjectFactory',
+  function(AnswerGroupObjectFactory, InteractionObjectFactory,
+    ContentObjectFactory) {
+    var State = function(name, classifierModelId, contentBackendList,
```

Comment:
Would it make more sense for this to accept a list of Content objects, an Interaction object, etc. -- and for the backend-->frontend conversion to happen in createFromBackendDict()? I'm thinking about objects like AnswerGroup which have two creation paths, one of which requires that conversion and the other does not.